src/lib.rs

```diff
@@ -1025,6 +1045,37 @@ impl<'a, K: 'a + Eq + Hash, V: 'a, S: 'a + BuildHasher + Clone> Map<'a, K, V, S>
         }
     }
 
+    fn _get_or_insert(&'a self, key: K, value: impl FnOnce(&K) -> V) -> Ref<'a, K, V, S> {
+        let hash = self.hash_usize(&key);
+
+        let idx = self.determine_shard(hash);
+
+        let shard = unsafe { self._yield_read_shard(idx) };
+
+        if let Some((kptr, vptr)) = shard.get_key_value(&key) {
+            unsafe {
+                let kptr: *const K = kptr;
+                let vptr: *const V = vptr.get();
+                Ref::new(shard, kptr, vptr)
+            }
+        } else {
+            drop(shard);
+
+            unsafe {
+                let c: K = ptr::read(&key);
```

Comment:
Here, K is duplicated illegally, can't do this. Violated standard rust rules and this also means K is dropped twice. If you use allocating string keys this segfaults.